packages/transformers/js/src/JSTransformer.js

```diff
@@ -939,14 +939,16 @@ export default (new Transformer({
 // also load the native module on the main thread, so that it is not unloaded until process exit.
 // See https://github.com/rust-lang/rust/issues/91979.
 let isLoadedOnMainThread = false;
+
+// $FlowFixMe
+let {glibcVersionRuntime} = process.report.getReport().header;
+
 async function loadOnMainThreadIfNeeded() {
   if (
     !isLoadedOnMainThread &&
```

Comment:
hmm it's surprising to me that this is running more than once per thread given this check here. any idea why that's happening?

Reply:
`isLoadedOnMainThread` is set to `true` only if — 

[> if(glibcVersionRuntime && parseFloat(glibcVersionRuntime) <= 2.17)](https://github.com/parcel-bundler/parcel/blob/db3bcae10497fa6a712fd9a135f93f26c5745454/packages/transformers/js/src/JSTransformer.js#L950-L957)

Else, its called once per transform —

https://github.com/parcel-bundler/parcel/blob/db3bcae10497fa6a712fd9a135f93f26c5745454/packages/transformers/js/src/JSTransformer.js#L308-L314

I think moving the check outside might be better in that case?

```ts
    if (glibcVersionRuntime && parseFloat(glibcVersionRuntime) <= 2.17) {
      let api = WorkerFarm.getWorkerApi();
      await api.callMaster({
        location: __dirname + '/loadNative.js',
        args: [],
      });
    }

   isLoadedOnMainThread = true;
```

Reply:
Ah right